sections/featured-collection.liquid

```diff
@@ -30,28 +30,32 @@
-    assign show_mobile_slider = true
+  assign show_slider_on_mobile = false
+  if section.settings.enable_mobile_slider and products_to_display > columns_mobile_int
+    assign show_slider_on_mobile = true
+  endif
+
+  assign show_slider_on_desktop = false
+  if section.settings.enable_desktop_slider and products_to_display > section.settings.columns_desktop
+    assign show_slider_on_desktop = true
   endif
 -%}
 
 <div class="color-{{ section.settings.color_scheme }} isolate gradient">
-  <div class="collection page-width{% if show_mobile_slider %} page-width-desktop{% endif %} section-{{ section.id }}-padding">
-    {%- unless section.settings.title == blank -%}
-      <div class="title-wrapper-with-link{% if show_mobile_slider %} title-wrapper--self-padded-tablet-down{% endif %} title-wrapper--no-top-margin">
-        <h2 class="title {{ section.settings.heading_size }}">{{ section.settings.title | escape }}</h2>
-  
-        {%- if section.settings.show_view_all and section.settings.swipe_on_mobile and more_in_collection -%}
-          <a href="{{ section.settings.collection.url }}" class="link underlined-link large-up-hide">{{ 'sections.featured_collection.view_all' | t }}</a>
+  <div class="collection section-{{ section.id }}-padding{% if section.settings.full_width %} collection--full-width{% endif %}">
+      <div class="collection__title title-wrapper title-wrapper--no-top-margin page-width{% if show_slider_on_mobile %} title-wrapper--self-padded-tablet-down{% endif %}">
+        {%- if section.settings.title != blank -%}
+          <h2 class="title {{ section.settings.heading_size }}">{{ section.settings.title | escape }}</h2>
+        {%- endif -%}
+        {%- if section.settings.description != blank or section.settings.show_description -%}
+          <div class="collection__description {{ section.settings.description_style }}">{%- if section.settings.show_description -%}{{ section.settings.collection.description }}{%- else -%}{{ section.settings.description }}{% endif %}</div>
         {%- endif -%}
       </div>
-    {%- endunless -%}
-  
-    <slider-component class="slider-mobile-gutter">
-      <ul id="Slider-{{ section.id }}" class="grid product-grid contains-card{% if settings.card_style == 'standard' %} contains-card--standard{% endif %} grid--{{ section.settings.columns_desktop }}-col-desktop {% if section.settings.collection == blank %} grid--2-col-tablet-down{% else %} grid--{{ section.settings.columns_mobile }}-col-tablet-down{% endif %}{% if show_mobile_slider %} slider slider--tablet grid--peek{% endif %}" role="list">
+
+    <slider-component class="slider-mobile-gutter{% if section.settings.full_width %} slider-component-full-width{% endif%}{% if show_slider_on_mobile == false %} page-width-mobile page-width-tablet{% endif %}{% if show_slider_on_desktop == false and section.settings.full_width == false %} page-width-desktop{% endif %}{% if show_slider_on_desktop %} collection__slider-desktop{% endif %}">
```

Comment:
```suggestion
    <slider-component class="slider-mobile-gutter{% if section.settings.full_width %} slider-component-full-width{% endif %}{% if show_slider_on_mobile == false %} page-width-mobile page-width-tablet{% endif %}{% if show_slider_on_desktop %} collection__slider-desktop{% else %}{% if section.settings.full_width == false %} page-width-desktop{% endif %}{% endif %}">
```

Missing a space `{% endif%}`. As for the last part, I find that easier to read (especially if you're getting rid of the `show_slider_on_desktop` and replacing it with two conditions) but no strong feelings about it. Feel free to keep as is :D